_drafts/making-backend-scalable.markdown

```diff
@@ -0,0 +1,130 @@
+---
+layout: post
+title: "Making Autolab's Backend Scalable"
+author: Ilter Canberk
+
+tags: [tango, redis, scalability]
+
+---
+
+[Tango](https://github.com/autolab/Tango) is a stand-alone, RESTful service that Autolab uses as the back-end for autograding. Tango receives grading jobs from Autolab's front-end, adds them in a job queue, assigns them to available containers for grading and shepherds the job throuout the process. In its early days, Tango was mostly used for jobs that ran in under 5 seconds. However, over the recent semesters, Autolab has grown to host classes like Distributed Systems, Machine Learning. Some of these classes have assessments that are computationally intensive on large datasets and contains bigger files.
+
+
+![Tango's initial Architecture]({{site-url}}/assets/redis1.png)
```

Comment:
- You mean to use `{{ site.url }}`.
- If (as Mihir mentioned) these diagrams really were made with Google Draw, it'd be awesome if you could Download as SVG from the "File" menu.
  - SVGs can have transparent backgrounds
  - SVG take up less space and achieve higher resolutions with no effort
  - it looks like you just screenshot'ed these (redis1.png below shows the red, dotted underline from a misspelled word)